.github/workflows/nightly.yml

```diff
@@ -1,9 +1,10 @@
 name: Nightly
 
 on:
-  schedule:
-    # every night at midnight
-    - cron:  '0 0 * * *'
+  [pull_request]
+#   schedule:
+#     # every night at midnight
+#     - cron:  '0 0 * * *'
```

Comment:
I don't understand that part, this should run one time a day to publish a nightly. I don't understand why it depends now on a PR?!

Reply:
Hi, thanks for pointing it out. As mentioned in the description of commit message, I made this change to make sure the nightly doesn't fail when we run it. It is more like cross-checking, if the current changes are merged then the nightly really works.
So I'll explain again if anything is unclear:

* In nightly we run 8 jobs(out of which currently the one for API 29 fails)
* So with the current changes(in this PR) and via _simulating_ nightly in this PR, I make sure if we run all those jobs, then the one for API29 works and any other job does not break/fail
 
I'll revert this change after reviewer(s) is(are) happy with the CI status. Let me know if I was unclear explaining it, will be happy to rectify and be more specific
PS: I agree, I should've written a comment in the code there mentioning the same reason, apologies for missing it : (